pkg/planner/core/integration_test.go

```diff
@@ -175,6 +175,196 @@ func TestPartitionPruningForEQ(t *testing.T) {
+		"TableReader_7 10.00 root partition:p2 data:Selection_6",
+		"└─Selection_6 10.00 cop[tikv]  eq(test.t_ts.report_updated, 2024-12-01 00:00:00)",
+		"  └─TableFullScan_5 10000.00 cop[tikv] table:t_ts keep order:false, stats:pseudo"))
+	tk.MustQuery("explain select * from t_ts where report_updated = '2024-12-01 00:00:00'").Check(testkit.Rows(
+		"TableReader_7 10.00 root partition:p2 data:Selection_6",
+		"└─Selection_6 10.00 cop[tikv]  eq(test.t_ts.report_updated, 2024-12-01 00:00:00.000000)",
+		"  └─TableFullScan_5 10000.00 cop[tikv] table:t_ts keep order:false, stats:pseudo"))
+	tk.MustQuery("explain select * from t_ts where report_updated > unix_timestamp('2008-05-01 00:00:00')").Check(testkit.Rows(
+		"TableReader_7 8000.00 root partition:all data:Selection_6",
+		"└─Selection_6 8000.00 cop[tikv]  gt(cast(test.t_ts.report_updated, double BINARY), 1.2095712e+09)",
+		"  └─TableFullScan_5 10000.00 cop[tikv] table:t_ts keep order:false, stats:pseudo"))
+	//MysQL explain:
+	//+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+	//| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra       |
+	//+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+	//|  1 | SIMPLE      | t_ts  | p1,p2,pm   | ALL  | NULL          | NULL | NULL    | NULL |    3 |    33.33 | Using where |
+	//+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+
+	// case 3: hash partition
+	tk.MustExec(`CREATE TABLE t_hash(a int, b int) PARTITION BY HASH(a) PARTITIONS 6`)
+	tk.MustExec(`insert into t_hash values(1, 1), (10, 10), (26, 26)`)
+	tk.MustQuery(`select * from t_hash where a = '1'`).Check(testkit.Rows("1 1"))
+	tk.MustQuery(`explain select * from t_hash where a = '1'`).Check(testkit.Rows(
+		"TableReader_7 10.00 root partition:p1 data:Selection_6",
+		"└─Selection_6 10.00 cop[tikv]  eq(test.t_hash.a, 1)",
+		"  └─TableFullScan_5 10000.00 cop[tikv] table:t_hash keep order:false, stats:pseudo"))
+	// MySQL explain:
+	//+----+-------------+--------+------------+------+---------------+------+---------+------+------+----------+-------------+
+	//| id | select_type | table  | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra       |
+	//+----+-------------+--------+------------+------+---------------+------+---------+------+------+----------+-------------+
+	//|  1 | SIMPLE      | t_hash | p1         | ALL  | NULL          | NULL | NULL    | NULL |    1 |   100.00 | Using where |
+	//+----+-------------+--------+------------+------+---------------+------+---------+------+------+----------+-------------+
+
+	// case 4: range columns partition
+	tk.MustExec(`CREATE TABLE t_range_col(a int, b int)
+		PARTITION BY RANGE COLUMNS(a) (
```

Comment:
I would use a different datatype if testing COLUMNS partitioning, since that is the main usage (apart from partitioning on multiple columns, which is not as useful).